public/js/index.js

```diff
@@ -2864,18 +2865,52 @@ function initTopicbar() {
             throttle: 500,
             cache: false,
             onResponse: function(res) {
-                var formattedResponse = {
+                let formattedResponse = {
                     success: false,
                     results: [],
                 };
+                const escapeHtml = function (text) {
+                    let esc = document.createElement('div');
+                    esc.innerHTML = text;
```

Comment:
User-generated content should *never* be fed to `.innerHTML` for security reasons. If the goal of this is to escape HTML entities, use a function like https://stackoverflow.com/a/6234804/808699.

Reply:
I previously used the same function as here ('https://github.com/go-gitea/gitea/pull/7225/files#diff-e70fdf29cba0fbb5a52462aed7e5c97dR3193') [1] which does the same thing as your linked method[2].
- the result: semantic js somehow parsed the content back to html code and inserted it with html element tags in preview. 
 As example i added your linked escape function. My inserted text is `<font style='color:green!important;'>colored</font>`
![Peek 2019-06-29 19-11](https://user-images.githubusercontent.com/24939127/60387349-e68e2c00-9aa1-11e9-8674-dc76970f760e.gif)
  As style is available, js is too.
- Therefore i used function above to escape and strip off html tags. (special letters are rejected on topics by gitea server). It is safe if the element `esc` is never added to DOM. Pro: browser could be used for parsing html. 
- I could change this to jquery implementation mentioned here: https://stackoverflow.com/questions/5002111/how-to-strip-html-tags-from-string-in-javascript/5002157#5002157 [4]
 but this does the exactly same thing:
![grafik](https://user-images.githubusercontent.com/24939127/60387455-d37c5b80-9aa3-11e9-9464-333042a20f74.png)
   - esc1: [1]
   - esc2: [2]
   - esc3: current implementation, without jquery
   - esc4: [4]

Reply:
If it's never added to the DOM, it should be safe. Looking at our .innerHTML usage, I'm pretty sure we have a few XSS vulnerabilities, some probably persistent. I always get suspious on any `.innerHTML` or `.html()` usage, almost always there are always better solutions.

If you just want to strip HTML tags from a string, this should be sufficient:

```js
function stripHTMLTags (str) {
  return str.replace(/<[^>]*>?/gm, "");
}

Reply:
Actually, I do have to correct myself. Even if HTML is not appended to the DOM, scripts can still execute. Try this in your browser console:

```js
let div = document.createElement('div');
div.innerHTML = `<img src='x' onerror='alert("xss")'>`;

So anyways, would prefer if you can switch to a regex-only solution.

Reply:
there is always something new to know, I didn't know this case.
I will replace it.